### PR TITLE
Fix passkey redirect

### PR DESCRIPTION
**Pull request by adriansalamon**
When logging in via passcode, a `fetch` request would be made to `/login/passkey/finish`, which in turn would redirect to the callback URL, resulting in a CORS error.

This fixes this by just updating the session cookie from this request. When the browser then redirects to `/`, the user will then already be logged in and redirected to the callback URL. Also renamed the cookie values to something more unique to logout :)

### CODE REVIEW

**Comment by foodelevator** [comment]
Ah, I had noticed that stuff didn't always work. Though I guess it only worked on the second try when I was already logged in... Nice fix!
I'm kind of inclined to prefer either adding a flag to `LoginUser` saying if it should redirect or not, or adding `redirect: "manual"` to the `fetch` call in `services/passkey/passkey.templ`, since otherwise it seems like `LoginUser` always should be used when logging users in. But idk, maybe that's weird, what do you think @adriansalamon?

File: services/user/auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package auth
+
+import "net/http"
+
+const SESSION_COOKIE string = "_logout_session"
+
+func UserCookie(sessionID string) *http.Cookie {
```
Comment (foodelevator) [inline services/user/auth/auth.go:7]:
```suggestion
func SessionCookie(sessionID string) *http.Cookie {
```

File: services/user/user.go [hunk 2]
```diff
@@ -70,20 +71,13 @@ func (s *service) LoginUser(ctx context.Context, kthid string) httputil.ToRespon
...
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		http.SetCookie(w, &http.Cookie{
-			Name:     "session",
-			Value:    sessionID.String(),
-			Path:     "/",
-			HttpOnly: true,
-			Secure:   true,
-			SameSite: http.SameSiteLaxMode,
-		})
+		http.SetCookie(w, auth.UserCookie(sessionID.String()))
```
Comment (foodelevator) [inline services/user/user.go:74]:
```suggestion
		http.SetCookie(w, auth.SessionCookie(sessionID.String()))
```

**Review by foodelevator — CHANGES_REQUESTED** [review]
Just some small naming thingies

**Comment by adriansalamon** [comment]
> Ah, I had noticed that stuff didn't always work. Though I guess it only worked on the second try when I was already logged in... Nice fix! I'm kind of inclined to prefer either adding a flag to `LoginUser` saying if it should redirect or not, or adding `redirect: "manual"` to the `fetch` call in `services/passkey/passkey.templ`, since otherwise it seems like `LoginUser` always should be used when logging users in. But idk, maybe that's weird, what do you think @adriansalamon?

There is also [`HX-Redirect`](https://htmx.org/headers/hx-redirect/) that we could use?

**Comment by foodelevator** [comment]
> There is also [`HX-Redirect`](https://htmx.org/headers/hx-redirect/) that we could use?

Unfortunately not, since that request is made with fetch directly (since you need some custom javascript to call `navigator.credentials.get` anyways.

**Review by foodelevator — APPROVED** [review]
Anyways, very nice changes, thanks!

**Pull request merged**